src/applications/facility-locator/components/FacilityTypeDropdown.jsx

```diff
@@ -22,11 +22,45 @@ class FacilityTypeDropdown extends Component {
     this.state = {
       showBubble: false,
     };
+
+    // eslint-disable-next-line prettier/prettier
+    this.setCCInfoRef = (element) => {
+      this.ccInfoIcon = element;
+    };
+    this.focusCCInfoIcon = () => {
+      if (this.ccInfoIcon) {
+        if (this.ccInfoCloseBtn) {
+          this.ccInfoCloseBtn.blur();
+        }
+        this.ccInfoIcon.focus();
+      }
+    };
+
+    // eslint-disable-next-line prettier/prettier
+    this.setCCInfoCloseRef = (element) => {
+      this.ccInfoCloseBtn = element;
+    };
+    this.focusCCInfoCloseBtn = () => {
+      if (this.ccInfoCloseBtn) {
+        this.ccInfoIcon.blur();
+        this.ccInfoCloseBtn.focus();
+      }
+    };
+  }
+
+  componentDidUpdate() {
+    // Set focus on the correct element (508)
+    if (this.state.showBubble) {
+      this.focusCCInfoCloseBtn();
+    } else {
+      this.focusCCInfoIcon();
+    }
   }
 
   // eslint-disable-next-line prettier/prettier
   toggleCCInfo = (e) => {
     e.preventDefault();
+
     this.setState({
       showBubble: !this.state.showBubble,
     });
```

Comment:
I tried out adding the following function call, and it focused the "i" button again as expected:

```javascript
  // Refocus the "i" button
  this.focusCCInfoIcon();
```

Reply:
Yes, add the call to `focusCCInfoIcon` after the `setState` declaration. So actually on line 67 or 68 if you leave a space.

That gives us the right state for the next interaction, and then sets focus correctly as a second discrete action.

Reply:
Recommend adding the call to focusCC in the callback param for the setState above, to guarantee it only runs after setState completes.